### PR TITLE
Symfony messenger implementation

### PR DESCRIPTION
**Pull request by cv65kr**
This PR contains:
- Bumped composer packages eg. Symfony 4.3 => 4.4
- Removed some deprecations
- Bumped docker-compose version
- Used Symfony Messenger Bus instead league tactician-bundle

### CODE REVIEW

**Comment by cv65kr** [comment]
@jorge07 Did You know how resolve problem with tests?

**Comment by jorge07** [comment]
Hi @cv65kr here the issue:
```
ERROR: Version in "./docker-compose.yml" is unsupported. You might be seeing this error because you're using the wrong Compose file version. Either specify a supported version (e.g "2.2" or "3.3") and place your service definitions under the `services` key, or omit the `version` key and place your service definitions at the root of the file to use version 1.

```

File: config/packages/twig.yaml [hunk 1]
```diff
@@ -2,3 +2,4 @@ twig:
     paths: ['%kernel.project_dir%/src/UI/Http/Web/templates']
     debug: '%kernel.debug%'
     strict_variables: '%kernel.debug%'
+    exception_controller: ~
```
Comment (jorge07) [inline config/packages/twig.yaml]: This param it's deprecated and I think it can be omitted ;)

File: src/Infrastructure/Share/MessageBusHelper.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+declare(strict_types=1);
+
+namespace App\Infrastructure\Share;
+
+use Symfony\Component\Messenger\Exception\HandlerFailedException;
+use Symfony\Component\Messenger\MessageBusInterface;
+use Symfony\Component\Messenger\Stamp\HandledStamp;
+use Throwable;
+
+final class MessageBusHelper
```
Comment (jorge07) [inline src/Infrastructure/Share/MessageBusHelper.php]: The idea of statics around the project it's something I don't like too much to be honest.What do you think about  have our own `CommandBus` and `QueryBus` objects injecting the required MessageBusInterface and handling the differences between one and the other independently (returning result and not). A Trait for the exception handling part should be enough to be DRY.
How does it sounds to you?

**Comment by cv65kr** [comment]
@jorge07 

I fixed docker issue yesterday, but there is a problem about tests which I describe in PR description.

If You look at `App\Tests\Application\ApplicationTestCase` we inject:

```
        $this->commandBus = $this->service('messenger.bus.command');
        $this->queryBus = $this->service('messenger.bus.query');
```

That's service are aliases for message buses, but this generate exceptions like below.

```
12) App\Tests\UI\Http\Rest\Controller\Events\GetEventsControllerTest::given_invalid_limit_returns_400_status
Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException: The "messenger.bus.command" service or alias has been removed or inlined when the container was compiled. You should either make it public, or stop using the container directly and use dependency injection instead.
```

Did You have any idea how resolve this problem?

File: src/Infrastructure/Share/MessageBusHelper.php (hunk 2 above)
Comment (cv65kr) [reply src/Infrastructure/Share/MessageBusHelper.php]: Sounds great for me, let's do this!

**Comment by jorge07** [comment]
hey @cv65kr just migrated to github actions as are way faster. Can you rebase please?

File: config/services.yaml [hunk 3]
```diff
@@ -14,26 +14,21 @@ services:
         public: false
         bind:
           $elasticConfig: '%elastic%'
-
-          $queryBus: '@tactician.commandbus.query'
-          $commandBus: '@tactician.commandbus.command'
```
Comment (jorge07) [inline config/services.yaml]: keeping this you can avoid this https://github.com/jorge07/symfony-4-es-cqrs-boilerplate/pull/132/files#diff-55b5b167120840a9198b5779c9b1de66R42-R48 just by using the var name

File: src/Infrastructure/Share/Bus/QueryBus.php [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        $this->messageBus = $messageBus;
+    }
+
+    /**
+     * @throws Throwable
+     */
+    public function handle($query)
+    {
+        try {
+            $command = $this->messageBus->dispatch($query);
```
Comment (jorge07) [inline src/Infrastructure/Share/Bus/QueryBus.php]: imo `$envelope` fits better than `$command` here

File: config/services.yaml (hunk 3 above)
Comment (cv65kr) [reply config/services.yaml]: This is not necessary, we want use message buses only by `QueryBus` and `CommandBus` classes.

So new declaration forces for us to use this services.

**Comment by cv65kr** [comment]
Rebased.

> I think you need to define this services as public in the services_test.yaml

I tried this but not working.

File: config/services.yaml (hunk 3 above)
Comment (jorge07) [reply config/services.yaml]: Yep. You're right

File: tests/UI/Http/Rest/Controller/JsonApiTestCase.php [hunk 5]
```diff
@@ -32,15 +32,15 @@ protected function createUser(string $email = self::DEFAULT_EMAIL, string $passw
             $password
         );
 
-        /** @var CommandBus $commandBus */
-        $commandBus = $this->cli->getContainer()->get('tactician.commandbus.command');
+        /** @var MessageBusInterface $commandBus */
+        $commandBus = $this->cli->getContainer()->get('messenger.bus.command');
```
Comment (jorge07) [inline tests/UI/Http/Rest/Controller/JsonApiTestCase.php]: Replace messenger bus in favor of the infrastructure one. This one shouldn't be public in the container. It fact it isn't and it's the reason why tests fails I think

Comment (cv65kr) [reply tests/UI/Http/Rest/Controller/JsonApiTestCase.php]: @jorge07 tests passed for now, but there is a problem with Github actions can You look at this?

**Comment by jorge07** [comment]
Awesome work here @cv65kr. Tests are failing due to depreciation but feel free to edit phpunit config to avoid this depreciations for now. We can address it in another PR along with the symfony 5 migration

**Comment by jorge07** [comment]
One last question that I've regarding this symfony messenger switch is if we still need sncrabbitmq

**Comment by cv65kr** [comment]
> One last question that I've regarding this symfony messenger switch is if we still need sncrabbitmq

For now yes, currently is related with Broadway stuff.

Green light from GA 👍

**Comment by jorge07** [comment]
LGTM

Will do a full qa this night and if all goes well will finally merge. 

Once again. @cv65kr thanks for the great work here.

Once merged I'll raise some issues to remove snc redis and have a bridge from Broadway to messenger, and another issue to start the v5 migration (may I rename the repository 😂)

File: tests/UI/Cli/Command/CreateUserCommandTest.php [hunk 6]
```diff
@@ -42,12 +42,12 @@ public function command_integration_with_bus_success(): void
         $this->assertStringContainsString('User Created:', $output);
         $this->assertStringContainsString('Email: jorge.arcoma@gmail.com', $output);
 
-        /** @var Item $item */
-        $item = $this->ask(new FindByEmailQuery($email));
+        $result = $this->ask(new FindByEmailQuery($email));
```
Comment (Lutacon) [inline tests/UI/Cli/Command/CreateUserCommandTest.php]: Why did you remove the comment?

File: tests/UI/Http/Rest/Controller/JsonApiTestCase.php [hunk 7]
```diff
@@ -70,7 +70,10 @@ protected function auth(string $username = self::DEFAULT_EMAIL, string $password
             '_password' => $password ?: self::DEFAULT_PASS,
         ]);
 
-        $response = json_decode($this->cli->getResponse()->getContent(), true);
+        /** @var string $content */
+        $content = $this->cli->getResponse()->getContent();
+
+        $response = json_decode($content, true);
```
Comment (Lutacon) [inline tests/UI/Http/Rest/Controller/JsonApiTestCase.php]: Is this really needed?

File: tests/Application/Query/User/FindByEmail/FindByEmailHandlerTest.php [hunk 8]
```diff
@@ -27,12 +27,12 @@ public function query_command_integration(): void
 
         $this->fireTerminateEvent();
 
-        /** @var Item $item */
-        $item = $this->ask(new FindByEmailQuery($email));
+        $result = $this->ask(new FindByEmailQuery($email));
+
```
Comment (Lutacon) [inline tests/Application/Query/User/FindByEmail/FindByEmailHandlerTest.php]: Why did you remove the comment?

File: src/Infrastructure/Share/Bus/CommandBus.php [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+use Symfony\Component\Messenger\MessageBusInterface;
+use Throwable;
+
+final class CommandBus
+{
+    use MessageBusExceptionTrait;
+
+    /**
+     * @var MessageBusInterface
+     */
```
Comment (Lutacon) [inline src/Infrastructure/Share/Bus/CommandBus.php]: Could this be in one line? 

```
/** @var MessageBusInterface */

File: tests/UI/Http/Rest/Controller/JsonApiTestCase.php (hunk 7 above)
Comment (cv65kr) [reply tests/UI/Http/Rest/Controller/JsonApiTestCase.php]: @Lutacon Thanks for review, appreciated that.

Yes this is necessary because `getContent()` return string, so we need decode this to get token.

File: src/Infrastructure/Share/Bus/CommandBus.php (hunk 9 above)
Comment (cv65kr) [reply src/Infrastructure/Share/Bus/CommandBus.php]: Don't worry about this, that's should be fixes in all places by #135.

**Comment by jorge07** [comment]
@cv65kr thanks you so much for the great work on this. I did a QA locally and all runs perfectly.  @Lutacon thanks for your review!
